instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/sessionreplay/SrSnapshotTest.kt

```diff
@@ -19,9 +22,12 @@ internal abstract class SrSnapshotTest :
         val instrumentation = InstrumentationRegistry.getInstrumentation()
         // we need this to avoid the Bitrise flakiness and to force and to wait for
         // SurfaceFlinger to call the onDraw method which will trigger the screen snapshot.
-        mockServerRule.activity.clickMeButton.performClick()
-
+        onView(ViewMatchers.isRoot()).perform(waitFor(UI_THREAD_DELAY_IN_MS))
         instrumentation.waitForIdleSync()
         return mockServerRule.activity.getExpectedSrData()
     }
+
+    companion object {
+        const val UI_THREAD_DELAY_IN_MS = 3000L
```

Comment:
I'm not super familiar with Espresso, but does this induce a 3 second delay on the test?
If yes, it seems like a really long time. Can it be reduced?

Reply:
yes...I wanted to make sure that this activity is not killed by the instrumentation very fast...it is just for this case where we are testing the snapshot capture for SR , we have a debouncer there that sometimes doesn't have the time to step in if the activity is killed very fast.